proposals/0003.md

```diff
@@ -0,0 +1,171 @@
+simple statement that you, as a contributor, have the legal right to make the
+contribution.
+
+In order to show your agreement with the DCO you should include at the end of the commit message,
+the following line: Signed-off-by: John Doe <john.doe@example.com>, using your real name and email.
+
+This can be done easily using the -s flag on the git commit.
+
+If you find yourself pushed a few commits without Signed-off-by, you can still add it afterwards.
+
+### Fixing the DCO
+
+If you did not sign-off one or more of your commits then it is not all for not. The
+crowd at `src-d` have a wonderful guide on how to remedy this situation
+[here](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository)
+
+## License Header
+
+As part of our license, we must include a license header at the top of each
+source file. The template for this header can be found
+[here](./path/to/license-header). If you are creating a new file in the
+repository you will have to add this header to the file.
+```
+
+## DCO Bot
+
+To ensure that all commits by an outside contributor are signed-off we SHALL add a
+[DOC bot][dco-bot] to monitor pull requests on our repositories.
+
+## Merging Pull Requests
+
+We SHALL NOT merge pull requests using the GitHub UI. This is to avoid having
+GitHub as a contributor and to ensure that our developers are adding a sign-off
+to the squash commit.
+
+Our maintainers should follow the following checklist when merging a pull
+request:
+- [ ] Has the PR received the appropriate number of approvals?
+- [ ] Has the CI finished and passed all the tests?
+- [ ] Does the DCO bot say all commits have been signed-off?
+
+This will be indicated by the green checks in the GitHub UI. Once you are
+satisfied then you can use:
+
+```
+$ git merge -s --squash
+```
+
+Since this commit could contain several authors it is RECOMMENDED that you do a
+double-check for all sign-offs.
+
+At this point you SHALL add a final sign-off:
+
+```
+Signed-off-by: John Doe <john.doe@example.com>
+```
+
+## Monadic Sign-offs
+
+While it is not strictly necessary, we are RECOMMENDING that all maintainers
+from Monadic sign-off on their commits as well. The easiest way to set this is
+up by including it a `git commit` alias, for example:
+
+```
+# gcs for git-commit-sign-off
+alias gcs="git commit -s"
+```
+
+## Pull Request Template
+
+It is RECOMMENDED that the repository contains [pull request
+template][pr-template] which has the following checklist:
+
+```
+- [ ] Have you [signed-off](./path/to/CONTRIBUTING.md) on all your commits?
+- [ ] Did you include the [license header](./path/to/CONTRIBUTING.md) in any new
+  files you created?
+```
+
+# Open Questions
+
+* How do we enforce CI has passed before merging into the main-line branch?
+* Do we use our @monadic.xyz emails or do we use our GitHub emails?
```

Comment:
Using our @monadic.xyz isn't absolutely critical - but it makes things more clear and consistent from a legal perspective.